File: eBarberShop.Models/Proizvod.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace eBarberShop.Models
{
    public class Proizvod
    {
        [Browsable(false)]
        public int ProizvodID { get; set; }
        public string Naziv { get; set; }
        public string Sifra { get; set; }
        public decimal Cijena { get; set; }
        public byte[] Slika { get; set; }
        [Browsable(false)]
        public int VrstaProizvodaID { get; set; }
        [Browsable(false)]
        public virtual VrstaProizvoda VrstaProizvoda { get; set; }
        public string VrstaProizvodaNaziv => VrstaProizvoda?.Naziv;
    }
}

File: ebarbershop_mobile/lib/models/proizvod.dart
import 'package:ebarbershop_mobile/models/vrstaproizvoda.dart';
import 'package:flutter/cupertino.dart';
import 'package:json_annotation/json_annotation.dart';

part 'proizvod.g.dart';

@JsonSerializable()
class Proizvod{
  int? proizvodID;
  String? naziv;
  String? sifra;
  double? cijena;
  String? slika;
  int? vrstaProizvodaID;
  VrstaProizvoda? vrstaProizvoda;

  Proizvod();

  factory Proizvod.fromJson(Map<String, dynamic> json) => _$ProizvodFromJson(json);

    Map<String, dynamic> toJson() => _$ProizvodToJson(this);
}